src/main/java/com/librato/disco/DiscoClient.java

```diff
@@ -124,10 +124,10 @@ public void stop() throws IOException {
 
     Node<T> _toNode(ChildData data) {
         String path = pathFromData(data);
-        String[] split = path.split(":");
-        if (split.length != 2) {
-            throw new RuntimeException("Don't know how to parse node path: " + path);
-        }
+        // This is somewhat hacky support for ipv6 with the same host:port notation
+        int l = path.lastIndexOf(':');
```

Comment:
Check for < 0 here to protect against future registrations that may not have port?